UserInterface/ViewModels/TestManagementViewModel.cs

```diff
@@ -91,248 +101,378 @@ public int Selected
     #endregion
 
     #region Errors
-
-    public string this[string columnName]
+    private bool CheckValidityInput()
     {
-        get
+        // Check the validity of the input
+        string testNameValidation = ErrorService.ValidateInput("TestName", TestName!);
+        string audienceValidation = ErrorService.ValidateInput("Audience", Audience!);
+
+        if (!string.IsNullOrEmpty(testNameValidation))
+        {
+            OpenErrorModal(testNameValidation);
+            return false;
+        }
+
+        if (!string.IsNullOrEmpty(audienceValidation))
         {
-            string? validationMessage = null;
-
-            switch (columnName)
-            {
-                case "TestName":
-                    if (string.IsNullOrEmpty(TestName))
-                        validationMessage = ErrorMessageStore.ErrorTestName;
-                    else if(TestName.Contains(ErrorMessageStore.IllegalCharacters))
-                        validationMessage = ErrorMessageStore.ErrorIllegalCharacters;
-                    break;
-                case "Audience":
-                    if (Audience == null || string.IsNullOrEmpty(Audience.Label))
-                        validationMessage = ErrorMessageStore.ErrorAudience;
-                    break;
-                default:
-                    break;
-            }
-
-            return validationMessage ?? string.Empty;
+            OpenErrorModal(audienceValidation);
+            return false;
         }
+        return true;
     }
     #endregion
 
     public bool IsConfirmed { get; set; }
-    public ITest test { get; set; }
-    private ITargetAudience tempTargetAudience;
-
-    private ErrorModalViewModal _errorModalViewModal { get; set; }
-    private ConfirmationModalViewModel _confirmationModalViewModel { get; set; }
+    public Test Test { get; set; }
+    private ConfirmationModalViewModel confirmationModalViewModel { get; set; }
 
-    public TestManagementViewModel(NavigationStore navigationStore, TestOverviewViewModel testOverviewViewModel,ITargetAudience targetAudience, ITest test = null)
+    public TestManagementViewModel(NavigationStore navigationStore, Test? test = null)
     {
         //Dependencies initialization
-        _testRepository = new TestRepository();
-        _targetAudienceRepository = new TargetAudienceRepository();
-        _testSerivce = new TestService(_testRepository);
-        _targetAudienceSerivce = new TargetAudienceService(_targetAudienceRepository);
-        _testOverviewViewModel = testOverviewViewModel;
-        _navigationStore = navigationStore;
-
-        tempTargetAudience = targetAudience;
+        this.navigationStore = navigationStore;
+
+        testRepository = new TestRepository();
+        targetAudienceRepository = new TargetAudienceRepository();
+
+        testService = new TestService(testRepository);
+        targetAudienceSerivce = new TargetAudienceService(targetAudienceRepository);
```

Comment:
ik geloof dat je de repositories niet direct in deze viewmodel gebruikt (zou ook niet moeten). Dan denk ik dat het beter is om ze ook niet als variabale te setten

Reply:
changed.